### PR TITLE
fix panic when Linux is nil

### PR DESCRIPTION
**Pull request by crosbymichael**
Linux is not always not nil.
If Linux is nil, panic will occur.

Closes #1449

This is a rebase of #1449

Signed-off-by: Ma Shimiao <mashimiao.fnst@cn.fujitsu.com>
Signed-off-by: Michael Crosby <crosbymichael@gmail.com>

### CODE REVIEW

**Comment by mrunalp** [comment]
LGTM

File: libcontainer/specconv/spec_linux.go [hunk 1]
```diff
@@ -566,53 +569,54 @@ func createDevices(spec *specs.Spec, config *configs.Config) error {
...
 		}
-		config.Devices = append(config.Devices, device)
 	}
 	return nil
 }
 
 func setupUserNamespace(spec *specs.Spec, config *configs.Config) error {
-	if len(spec.Linux.UIDMappings) == 0 {
-		return nil
-	}
```
Comment (dqminh) [inline libcontainer/specconv/spec_linux.go]: this removal seems unintended. i think if spec.Linux is nil here we shoud just bail to be the same as the old code.

Comment (crosbymichael) [inline libcontainer/specconv/spec_linux.go]: The nil check should be on 613 now

Comment (crosbymichael) [inline libcontainer/specconv/spec_linux.go]: nvm, fixed it

**Comment by cyphar** [comment]
LGTM.

**Comment by hqhq** [comment]
LGTM

**Pull request merged**